### PR TITLE
Added golint check for pkg/kubelet.

### LINKED ISSUE #47315: Add golint check for pkg/kubelet
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
enhancement/cleanup

**Kubernetes version** (use `kubectl version`):
master branch

**Description**:
It's better to add golint check for `pkg/kubelet`, e.g. checking comments, naming.

### PR DESCRIPTION
**Pull request by k82cn**
**What this PR does / why we need it**:
Added golint check for pkg/kubelet, and make golint happy.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #47315 

**Release note**:
```release-note-none
```

### CODE REVIEW

**Comment by k82cn** [comment]
WIP PR, I'll make golint happy in this PR :).

**Comment by ixdy** [comment]
/unassign
/assign @k82cn 

Please reassign when this is ready for review.

**Comment by k82cn** [comment]
sure, np :).

**Comment by cblecker** [comment]
👍🏻👍🏻 to better code comments :)

**Comment by k82cn** [comment]
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Comment by ixdy** [comment]
`hack/` changes look fine.
/approve

We definitely should have a reviewer in `pkg/kubelet` look at this, though.
/unassign @ixdy @fejta @k82cn

**Comment by k82cn** [comment]
@kubernetes/sig-node-pr-reviews

File: pkg/kubelet/reason_cache.go [hunk 1]
```diff
@@ -40,17 +40,18 @@ type ReasonCache struct {
 	cache *lru.Cache
 }
 
-// reasonInfo is the cached item in ReasonCache
-type reasonInfo struct {
-	reason  error
-	message string
+// Reason is the cached item in ReasonCache
+type Reason struct {
```
Comment (vishh) [inline pkg/kubelet/reason_cache.go]: Why is it public?

Comment (k82cn) [inline pkg/kubelet/reason_cache.go]: golint report a warning that the `error` has to be the latest one of return value at https://github.com/kubernetes/kubernetes/pull/47316/files#diff-ade44c0052b07470763ef431d55210ceL95 .
Compare to moving `error` to the latest one (the latest return value indicates whether found reason in cache), so prefer to use an object as the result value.

**Comment by k82cn** [comment]
ping for approve & lgtm :).

**Comment by k82cn** [comment]
@vishh , do you have more comments?

**Comment by k82cn** [comment]
@vishh , do you have more comments?

File: pkg/kubelet/kubelet.go [hunk 2]
```diff
@@ -696,14 +696,14 @@ func NewMainKubelet(kubeCfg *componentconfig.KubeletConfiguration, kubeDeps *Kub
...
+			if err != nil {
 				return nil, err
-			} else {
-				ips = localIPs
 			}
-		} else {
-			ips = []net.IP{cfgAddress}
+			ips = localIPs
 		}
+		ips = []net.IP{cfgAddress}
```
Comment (yujuhong) [inline pkg/kubelet/kubelet.go]: This line should only be run if `cfgAddress == nil || cfgAddress.IsUnspecified()` is not true. Now it always runs regardless.

File: pkg/kubelet/reason_cache.go (hunk 1 above)
Comment (yujuhong) [inline pkg/kubelet/reason_cache.go]: You still don't need to export `Reason` since it's only used in this package, right?

**Review by yujuhong — CHANGES_REQUESTED** [review]

File: pkg/kubelet/kubelet.go (hunk 2 above)
Comment (k82cn) [inline pkg/kubelet/kubelet.go]: wow, good catch :), thanks very much.

File: pkg/kubelet/reason_cache.go (hunk 1 above)
Comment (k82cn) [inline pkg/kubelet/reason_cache.go]: agree, renamed to `reasonItem` (reason is used by other var).

**Review by yujuhong — APPROVED** [review]

**Comment by yujuhong** [comment]
/lgtm

Thanks for cleaning this up!

**Pull request merged**